astropy/stats/src/fast_sigma_clip.c

```diff
@@ -106,7 +107,7 @@ static void _sigma_clip_fast(
     double *mad_buffer = NULL;
 
     // data_buffer is used to store the current values being sigma clipped
-    data_buffer = (double *)PyArray_malloc(n_i * sizeof(double));
+    data_buffer = malloc(n_i * sizeof(double));
```

Comment:
shouldn't we still cast to the declared type of `data_buffer` (keep `(double *)`) ?